vault/raft.go

```diff
@@ -473,9 +473,24 @@ func (c *Core) raftTLSRotatePhased(ctx context.Context, logger hclog.Logger, raf
 		return errors.New("no active raft TLS key found")
 	}
 
+	getNextRotationTime := func(next time.Time) time.Time {
+		now := time.Now()
+
+		// active key's CreatedTime + raftTLSRotationPeriod might be in
+		// the past (meaning it is ready to be rotated) which will cause
+		// NewTicker to panic when used with time.Until, prevent this by
+		// pushing out rotation time into very near future
+		if next.Before(now) {
+			return now.Add(1 * time.Minute)
```

Comment:
I think this is fine, though I think it's not quite what the behaviour was before, right?  Prior to the breaking change, I assume a negative next would be treated as "immediately", not "now+1m"?

Reply:
That's true but with `NewTicker` I think all you can really do is try to be as close to "immediately" as possible. The reason for adding 1m was just to try to add enough buffer to prevent the negative `NewTicker` issue. It needs to be some time in the future in other words.

Reply:
I reduced the padding to be 10 seconds. I'll admit that the 1 minute was a bit aggressive.